packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -691,5 +690,5 @@ Future<DateTime> showDatePicker({
       lastDate: lastDate
     )
   );
-  return picked ?? initialDate;
+  return picked ?? null;
```

Comment:
Fair enough. I thought the null return would appear more transparent and obviously intentional when written that way, but I can update :)